pkg/storage/unified/apistore/store.go

```diff
@@ -480,43 +491,45 @@ func (s *Storage) GuaranteedUpdate(
 			}
 		}
 
-		created = true
-		existingObj = s.newFunc()
-		if len(rsp.Value) > 0 {
-			created = false
-			_, err = s.convertToObject(rsp.Value, existingObj)
-			if err != nil {
-				return err
+		// Not found (should we create or error?)
+		if len(rsp.Value) == 0 {
```

Comment:
most of the changes in this are refactoring `Update()` so that it delegates to the same `Create` method when necessary